packs/faasd/templates/faasd.nomad.tpl

```diff
@@ -0,0 +1,292 @@
+      }
+      [[- end]]
+    }
+
+    [[- if .faasd.register_auth_consul_service ]]
+    service {
+      name = "[[ .faasd.auth_consul_service_name ]]"
+      [[if ne (len .faasd.consul_service_tags) 0 ]]
+      tags = [[ .faasd.consul_service_tags | toPrettyJson ]]
+      [[ end ]]
+      port = "auth_http"
+
+      check {
+        type     = "tcp"
+        port     = "auth_http"
+        interval = "5s"
+        timeout  = "2s"
+      }
+    }
+    [[- end ]]
+
+    [[- if .faasd.register_nats_consul_service ]]
+    service {
+      name = "[[ .faasd.nats_consul_service_name ]]"
+      [[- if ne (len .faasd.consul_service_tags) 0 ]]
+      tags = [[ .faasd.consul_service_tags | toPrettyJson ]]
+      [[- end ]]
+      port = "nats_tcp_client"
+
+      check {
+        type     = "tcp"
+        port     = "nats_tcp_client"
+        interval = "5s"
+        timeout  = "2s"
+      }
+    }
+    [[- end ]]
+
+    [[- if .faasd.register_nats_consul_service ]]
+    service {
+      name = "faasd-nats-monitoring"
+      [[- if ne (len .faasd.nats_consul_service_name) 0 ]]
+      tags = [[ .faasd.consul_service_tags | toPrettyJson ]]
+      [[- end ]]
+      port = "nats_http_mon"
+
+      check {
+        type     = "http"
+        path     = "/connz"
+        port     = "nats_http_mon"
+        interval = "30s"
+        timeout  = "2s"
+      }
+    }
+    [[- end]]
+
+    [[- if .faasd.register_gateway_consul_service ]]
+    service {
+      name = "[[ .faasd.gateway_consul_service_name ]]"
+      [[- if ne (len .faasd.consul_service_tags) 0 ]]
+      tags = [[ .faasd.consul_service_tags | toPrettyJson ]]
+      [[- end ]]
+      port = "gateway_http"
+
+      check {
+        type     = "http"
+        path     = "/healthz"
+        port     = "gateway_http"
+        interval = "5s"
+        timeout  = "2s"
+      }
+    }
+    [[- end ]]
+
+    [[- if .faasd.register_provider_consul_service ]]
+    service {
+      name = "[[ .faasd.provider_consul_service_name ]]"
+      [[if ne (len .faasd.consul_service_tags) 0 ]]
+      tags = [[ .faasd.consul_service_tags | toPrettyJson ]]
+      [[ end ]]
+      port = "faasd_provider_tcp"
+
+      check {
+        type     = "tcp"
+        port     = "faasd_provider_tcp"
+        interval = "5s"
+        timeout  = "2s"
+      }
+    }
+    [[- end ]]
+
+    restart {
+      attempts = 3
+      delay    = "5s"
+      interval = "10m"
+      mode     = "delay"
+    }
+
+    task "download-faasd" {
+      lifecycle {
+        hook    = "prestart"
+      }
+
+      driver = "raw_exec"
+      config {
+        command = "sh"
+        args    = ["-c", "wget -q https://github.com/openfaas/faasd/releases/download/[[.faasd.faasd_version]]/faasd && mkdir -p /var/lib/faasd && touch /var/lib/faasd/hosts /var/lib/faasd/resolv.conf && mv faasd /usr/local/bin/faasd && chmod +x /usr/local/bin/faasd"]
+      }
+      env {
+        service_timeout = "60s"
+      }
+    }
+
+    task "faasd_provider" {
```

Comment:
Yes, you're right! I'll update it with `prepare_faasd` , I think that is a better name for the task.

Reply:
Ok.  But I'm still curious if you plan to use `prepare-faasd` or `prepare_faasd`.  Notice the hypen instead of underscore? All the other tasks use hyphens like `basic-auth-plugin`.

Reply:
again, you're right. There are no particular reasons...actually, it could be useful to add some guidelines also here in the pack docs. Could be added to [this](https://github.com/hashicorp/nomad-pack-community-registry/issues/50) issue too.

Reply:
Regarding the tasks, I'll uniform every task name to "_".

Reply:
I'll definitely talk to the team about naming convention guidance. Thanks for pointing that out. I'll merge this in as soon as you update the names.  Again, thanks for submitting this!